### DISCUSSION TITLE
Tweak: Hide blocked reblog targets

### DISCUSSION THREAD

**Discussion by invalidCards** [AprilSylph/XKit-Rewritten#474] [Suggestion Box]
I would like to suggest a feature that can absolutely completely eradicate blocked users from the dashboard. Currently Tumblr already implements posts being hidden if any author(? or just the OP? I'm not quite sure) is blocked, and content filtering can remove posts that include tagging blocked people if it has been manually set up that way. Tumblr does not, however, hide posts that have been directly reblogged by one of your followers from a blocked blog, which is what I am suggesting here.

There [is an API endpoint to fetch a blog's blocks](https://github.com/tumblr/docs/blob/master/api.md#blocks--retrieve-blogs-blocks) but it can only fetch 20 blocks at a time at a rate of 60 chunks per minute, so I don't know how feasible it is to implement this into a tweak; alternatively, a text box could be provided within the XKit Rewritten settings interface allowing users to provide a comma-separated list of blogs to hide, though this once again borders on the (previously rejected) blocklist functionality which would not be ideal.

**Comment by marcustyphoon** [comment]
~~Wow, Tumblr seriously doesn't do filtered word checks on trail URLs either? That's... ridiculous.~~

I just checked, and blog objects in `timelineObject` data sadly don't contain `is_blocked_from_primary` fields, which would have made this super easy ;(

**Reply by AprilSylph** [reply]
It does. I think Tom is suggesting to hide posts based purely on the reblog parent, even if the OP isn't blocked nor has the reblog parent contributed to the post trail.

https://tumblr.zendesk.com/hc/en-us/articles/115015814708-Tag-and-Post-Content-Filtering
> The content filter includes all the text within a post, including the blog name and blog names within the reblog trail.

**Reply by marcustyphoon** [reply]
Weird, not working for me. I guess I'll bug report it.

Edit: Semi-never-mind; I was testing on the first post on my dash with a lengthy reblog chain, which was Cyle reblogging the thread of responses to a sponsored post. Evidently they can't be filtered. Carry on.

Anyways, the fact that Tumblr doesn't filter these is a clear bug, IMO.

**Comment by AprilSylph** [comment]
Tumblr's content filtering now checks reblog parent usernames in addition to usernames in the reblog trail.
